ddl/table.go

```diff
@@ -484,3 +484,73 @@ func checkAddPartitionValue(meta *model.TableInfo, part *model.PartitionInfo) er
 	}
 	return nil
 }
+
+func checkPartitionRangeList(meta *model.TableInfo, partName string) error {
+	if meta.GetPartitionInfo() == nil {
+		return errors.Trace(ErrPartitionMgmtOnNonpartitioned)
+	}
+	oldDefs := meta.Partition.Definitions
+	for _, def := range oldDefs {
+		if strings.EqualFold(def.Name, partName) {
+			if len(oldDefs) == 1 {
+				return ErrDropLastPartition
+			}
+			return nil
+		}
+	}
+	return errors.Trace(ErrDropPartitionNonExistent.GenByArgs(partName))
+}
+
+func dropPartitionAndPartitionInfo(job *model.Job, t *meta.Meta, tblInfo *model.TableInfo, partName string) error {
+	newDefs, oldDefs := []model.PartitionDefinition{}, tblInfo.Partition.Definitions
+	newDefs = make([]model.PartitionDefinition, 0, len(oldDefs)-1)
+	var pid int64
+	for i := 0; i < len(oldDefs); i++ {
+		if strings.EqualFold(oldDefs[i].Name, partName) {
```

Comment:
if !strings.EqualFold(oldDefs[i].Name, partName)  {
continue;
}
BTW: I think we should use CIStr for ldDefs[i].Name.